File: src/com/eagle/programmar/CSS/Values/CSS_URL_Value.java
// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Aug 12, 2022

package com.eagle.programmar.CSS.Values;

import com.eagle.programmar.CSS.Terminals.CSS_Base64;
import com.eagle.programmar.CSS.Terminals.CSS_FileName;
import com.eagle.programmar.CSS.Terminals.CSS_Keyword;
import com.eagle.tokens.TokenChooser;
import com.eagle.tokens.TokenSequence;
import com.eagle.tokens.punctuation.PunctuationColon;
import com.eagle.tokens.punctuation.PunctuationComma;
import com.eagle.tokens.punctuation.PunctuationLeftParen;
import com.eagle.tokens.punctuation.PunctuationRightParen;
import com.eagle.tokens.punctuation.PunctuationSemicolon;
import com.eagle.tokens.punctuation.PunctuationSlash;

public class CSS_URL_Value extends TokenSequence
{
	public @S(10) CSS_Keyword URL = new CSS_Keyword("url");
	public @S(20) PunctuationLeftParen leftParen;
	public @S(30) CSS_File file;
	public @S(40) PunctuationRightParen rightParen;

	public static class CSS_File extends TokenChooser
	{
		public @CHOICE CSS_FileName fileName;

		public @CHOICE static class CSS_FileInline extends TokenSequence
		{
			public @S(10) CSS_Keyword DATA = new CSS_Keyword("data");
			public @S(20) PunctuationColon colon;
			public @S(30) CSS_Keyword IMAGE = new CSS_Keyword("image");
			public @S(40) PunctuationSlash slash;
			public @S(50) CSS_Keyword PNG = new CSS_Keyword("png");
			public @S(60) PunctuationSemicolon semicolon;
			public @S(70) CSS_Keyword BASE64 = new CSS_Keyword("base64");
			public @S(80) PunctuationComma comma;
			public @S(90) CSS_Base64 base64;
		}
	}
}

File: csrc/com/eagle/programmar/CSS/Values/CSS_URL_Value.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// Purchase a Premium Edition license at:
// https://www.tangiblesoftwaresolutions.com/order/order-java-to-csharp.html
// ====================================================================================================

// Copyright Eagle Legacy Modernization LLC, 2010-date
// Original author: Steven A. O'Hara, Aug 12, 2022

namespace com.eagle.programmar.CSS.Values
{
	using CSS_Base64 = com.eagle.programmar.CSS.Terminals.CSS_Base64;
	using CSS_FileName = com.eagle.programmar.CSS.Terminals.CSS_FileName;
	using CSS_Keyword = com.eagle.programmar.CSS.Terminals.CSS_Keyword;
	using TokenChooser = com.eagle.tokens.TokenChooser;
	using TokenSequence = com.eagle.tokens.TokenSequence;
	using PunctuationColon = com.eagle.tokens.punctuation.PunctuationColon;
	using PunctuationComma = com.eagle.tokens.punctuation.PunctuationComma;
	using PunctuationLeftParen = com.eagle.tokens.punctuation.PunctuationLeftParen;
	using PunctuationRightParen = com.eagle.tokens.punctuation.PunctuationRightParen;
	using PunctuationSemicolon = com.eagle.tokens.punctuation.PunctuationSemicolon;
	using PunctuationSlash = com.eagle.tokens.punctuation.PunctuationSlash;

	public class CSS_URL_Value : TokenSequence
	{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.CSS.Terminals.CSS_Keyword URL = new com.eagle.programmar.CSS.Terminals.CSS_Keyword("url");
		public CSS_Keyword URL = new CSS_Keyword("url");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.tokens.punctuation.PunctuationLeftParen leftParen;
		public PunctuationLeftParen leftParen;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) CSS_File file;
		public CSS_File file;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.tokens.punctuation.PunctuationRightParen rightParen;
		public PunctuationRightParen rightParen;

		public class CSS_File : TokenChooser
		{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE CSS_FileName XXfileName;
			public CSS_FileName XXfileName;

// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @CHOICE static class CSS_FileInline extends com.eagle.tokens.TokenSequence
			public class CSS_FileInline : TokenSequence
			{
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(10) com.eagle.programmar.CSS.Terminals.CSS_Keyword DATA = new com.eagle.programmar.CSS.Terminals.CSS_Keyword("data");
				public CSS_Keyword DATA = new CSS_Keyword("data");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(20) com.eagle.tokens.punctuation.PunctuationColon colon;
				public PunctuationColon colon;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(30) com.eagle.programmar.CSS.Terminals.CSS_Keyword IMAGE = new com.eagle.programmar.CSS.Terminals.CSS_Keyword("image");
				public CSS_Keyword IMAGE = new CSS_Keyword("image");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(40) com.eagle.tokens.punctuation.PunctuationSlash slash;
				public PunctuationSlash slash;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(50) com.eagle.programmar.CSS.Terminals.CSS_Keyword PNG = new com.eagle.programmar.CSS.Terminals.CSS_Keyword("png");
				public CSS_Keyword PNG = new CSS_Keyword("png");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(60) com.eagle.tokens.punctuation.PunctuationSemicolon semicolon;
				public PunctuationSemicolon semicolon;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(70) com.eagle.programmar.CSS.Terminals.CSS_Keyword BASE64 = new com.eagle.programmar.CSS.Terminals.CSS_Keyword("base64");
				public CSS_Keyword BASE64 = new CSS_Keyword("base64");
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(80) com.eagle.tokens.punctuation.PunctuationComma comma;
				public PunctuationComma comma;
// JAVA TO C# CONVERTER TASK: Most Java annotations will not have direct .NET equivalent attributes:
// ORIGINAL LINE: public @S(90) com.eagle.programmar.CSS.Terminals.CSS_Base64 base64;
				public CSS_Base64 base64;
			}
		}
	}
}
